### PR TITLE
Words filter and User activities

### PR DESCRIPTION
**Pull request by huangoc93**

### CODE REVIEW

File: app/controllers/words_controller.rb [hunk 1]
```diff
@@ -2,6 +2,22 @@ class WordsController < ApplicationController
   before_action :logged_in_user
 
   def index
-    @words = Word.page params[:page]
+    if category = Category.find_by_id(params[:category])
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: remove `()`, and rename to params[:category_id]

File: app/controllers/words_controller.rb [hunk 2]
```diff
@@ -2,6 +2,22 @@ class WordsController < ApplicationController
...
 
   def index
-    @words = Word.page params[:page]
+    if category = Category.find_by_id(params[:category]) 
+      @words = category.words
+    else
+      @words = Word.all
+    end
+
+    if params[:sort].to_i == 1
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: Why don't send params sort with asc or desc.

File: app/models/activity.rb [hunk 3]
```diff
@@ -2,4 +2,6 @@ class Activity < ActiveRecord::Base
   belongs_to :user
 
   validates :description,  presence: true
+
+  default_scope -> {order created_at: :desc}
```
Comment (dieunb) [inline app/models/activity.rb]: remove

File: app/controllers/words_controller.rb [hunk 4]
```diff
@@ -2,6 +2,17 @@ class WordsController < ApplicationController
   before_action :logged_in_user
 
   def index
-    @words = Word.page params[:page]
+    if category = Category.find_by_id(params[:category_id])
```
Comment (huangoc93) [inline app/controllers/words_controller.rb]: cannot remove() here before if statement

Comment (dieunb) [inline app/controllers/words_controller.rb]: OK (y)

File: app/controllers/words_controller.rb [hunk 5]
```diff
@@ -2,6 +2,17 @@ class WordsController < ApplicationController
...
+      @words = category.words
+    else
+      @words = Word.all
+    end
+
+    @words = @words.order(name: params[:sort]) if !params[:sort].blank?
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: unless

File: app/controllers/words_controller.rb [hunk 6]
```diff
@@ -2,6 +2,17 @@ class WordsController < ApplicationController
...
+    @words = @words.page params[:page]
+
+    @sorter = [["a -> z", :asc], ["z -> a", :desc]]
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: Move to view

File: app/controllers/words_controller.rb [hunk 7]
```diff
@@ -2,6 +2,17 @@ class WordsController < ApplicationController
...
+    @category_filter = Category.all.collect {|c| [c.name, c.id]}
+    @word_type = {"learned" => 1, "not learned" => -1, "all" => 0}
```
Comment (dieunb) [inline app/controllers/words_controller.rb]: move to view

**Pull request merged**